src/libsyntax/parse/token.rs

```diff
@@ -455,3 +461,38 @@ pub fn is_op(tok: &Token) -> bool {
         _ => true,
     }
 }
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct LazyTokenStream(RefCell<Option<TokenStream>>);
```

Comment:
Is this meant to be used by macro authors? If so could you document with why and when.